pkg/collectors/virtualservices.go

```diff
@@ -0,0 +1,110 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"github.com/douglas-reid/istio-state-metrics/pkg/apis/networking/v1alpha3"
+	"github.com/douglas-reid/istio-state-metrics/pkg/client/clientset/versioned"
+	informers "github.com/douglas-reid/istio-state-metrics/pkg/client/informers/externalversions/networking/v1alpha3"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descVirtualServiceHostsName          = "istio_pilot_virtual_service_hosts"
+	descVirtualServiceHostsHelp          = "Information about Hosts in Pilot VirtualServices"
+	descVirtualServiceHostsDefaultLabels = []string{"VirtualService", "hosts"}
+
+	descVirtualServiceHosts = prometheus.NewDesc(
+		descVirtualServiceHostsName,
+		descVirtualServiceHostsHelp,
+		descVirtualServiceHostsDefaultLabels,
+		nil,
+	)
+
+	descVirtualServicesInfo = prometheus.NewDesc(
+		"istio_pilot_virtualService_info",
+		"Information about Pilot VirtualServices",
+		[]string{"VirtualService", "namespace"},
+		nil,
+	)
+
+	descVirtualServicesGateways = prometheus.NewDesc(
+		"istio_pilot_virtualService_gateways",
+		"Information about Gateways in Pilot VirtualServices",
+		[]string{"VirtualService", "gateways"},
```

Comment:
similar to above: let's use snake_case and add `namespace` to everything

Reply:
Fixed the snake case.. as above... virtual_service label is a concat of name and namespace, should we still add it?

Reply:
Yes. Let's do it. and split `virtual_service` out, so that `virtual_service` is `<name>` and `namespace` is `<namespace>`.

Reply:
Thanks...fixed